codec_specs.md

```diff
@@ -727,3 +728,24 @@ Codec Name: VobBtn Buttons
 
 Description: Based on [MPEG/VOB PCI packets](http://dvd.sourceforge.net/dvdinfo/pci_pkt.html). The file contains a header consisting of the string "butonDVD" followed by the width and height in pixels (16 bits integer each) and 4 reserved bytes. The rest is full [PCI packets](http://dvd.sourceforge.net/dvdinfo/pci_pkt.html).
 
+## Timecode Codec Mappings
+
+### T_QUICKTIME/SINGLE
+
+Codec ID: T_QUICKTIME/SINGLE
+
+Codec Name: QuickTime timecode track (continuous)
+
+Description: `T_QUICKTIME/SINGLE` is used to identify Timecode tracks as stored in QuickTime. The `QuickTime Timecode Sample Data` is stored within the Block's data section. For an explanation of `QuickTime Timecode Sample Data` read [QuickTime File Format Specification](https://developer.apple.com/library/content/documentation/QuickTime/QTFF/QTFFChap3/qtff3.html#//apple_ref/doc/uid/TP40000939-CH205-57421). In the context of Matroska, `T_QUICKTIME/SINGLE` tracks MUST only relate to exactly one `Cluster` which MUST have a `Cluster Timecode` of zero and exactly one `SimpleBlock` or `BlockGroup Element`. The storage of the `QuickTime Timecode Sample Data` with the `SimpleBlock` or `BlockGroup Element` allows the `QuickTime Timecode` to be mapped to the `Matroska Timecode` at a single point. The `QuickTime Timecode` value MAY then be evaluated at any point in the `Matroska Timecode` by using the initialised `Timecode Sample Description`, the single `QuickTime Timecode Sample Data` value, and the `Matroska Timecode`.
```

Comment:
British `initialised` or American `initialized`? The rest seems to me American.

Reply:
Two things I'd like to discuss:

> In the context of Matroska, `T_QUICKTIME/SINGLE` tracks MUST only relate to exactly one `Cluster` which MUST have a `Cluster Timecode` of zero and exactly one `SimpleBlock` or `BlockGroup Element`.

I honestly don't understand what this is supposed to mean, especially the verb "relate". Do you mean to say that for a track with the CodecID `T_QUICKTIME/SINGLE` there MUST be exactly one `BlockGroup` or `SimpleBlock`, that this singular block is stored in a cluster with `ClusterTimecode` set to 0, and that there MUST NOT be any other cluster in the file with `ClusterTimecode` set to 0?

If yes, why such severe restrictions? Why not allow multiple `Clusters` that start at 0 (by simply omitting the restriction)? What about files whose first cluster doesn't start at 0 — how would that be interpreted? And why are you saying something about the `Cluster Timecode` but not about the timecode parts of the blocks?

In other words: can you please describe all of your thoughts in much more detail? We can then see how to turn that into unambiguous verbiage.

Reply:
The second issue I have is with linking to other web pages as part of the format specification. I'd rather we refer to existing PDF documentation or publications. In the case of linking to the QuickTime specification this is both possible and a bit difficult: possible because there is [a PDF version](https://developer.apple.com/standards/qtff-2001.pdf) of the QuickTime specs, and harder because it doesn't contain numbered chapters and sections. I still prefer to say something like this:

> For an explanation of the format read the section "Timecode Media", sub-section "Timecode Sample Description", of Apple's [QuickTime File Format Specification (2001)](https://developer.apple.com/standards/qtff-2001.pdf).

Reply:
In this case, I suggest that `T_QUICKTIME/SINGLE` implies that only a single Cluster is associated with that track. I didn't mean that imply that other tracks could not share the same timecode value. Upon reflection I think the requirement to use a Cluster Timecode is unnecessary. What is needed in T_QUICKTIME/SINGLE is for a single `QuickTime Timecode Sample Data` value to be stored in a single `BlockGroup` or `SimpleBlock` of a single `Cluster`, then the `QuickTime Timecode Sample Data` (within the Block) and the `Timecode Sample Description` (in Private Data) and the `Cluster Timecode` and the `Block Timecode` can be used to calculate what the QuickTime Timecode value should be relative to any Matroska Timecode value. Still for a Matroska Reader that needs to quickly associate any Matroska Timecode value with a QuickTime Timecode value it would be good to know where that Cluster is quickly. I had intended to have Cluster as 0 timecode to mean that it should be one of the first Clusters, but of course that is not a good expectation.

Reply:
If finding the (single) `BlockGroup`/`SimpleBlock` is the goal, then just say that the `BlockGroup`/`SimpleBlock` SHOULD be referenced via `Cues`.

I don't like requiring that the first or one of the first cluster contains that data as that seems problematic wrt streaming Matroska files.

Reply:
Are there any other Codec Mappings that require Cues?

Reply:
None that I know of.